### DISCUSSION TITLE
Custom WebService - Libraries javax.jws.*

### DISCUSSION THREAD

**Discussion by FTK7** [nextgenhealthcare/connect#5153] [Q&A] [answered]
Hello Everyone, 

I am trying to develop a custom web service for Mirth 4.0.0. It seems quite simple to only change the name of the method, but I can't compile the file.
Everywhere I have found they use the libraries javax.jws.\*. I have created an Eclipse project and it keeps saying it's unable to resolve them.
![imatge](https://user-images.githubusercontent.com/66680696/165259711-04cb3320-02e6-4551-8705-4fd8c2bd4ba8.png)

I have been searching for it, and I think it is related to java versions, but I am unable to get it.
Somewhere I have found it was replaced by jakarta.jws.* but in the source code of mirth i see it uses javax.jws.*
https://github.com/freemed/mirth/blob/master/server/src/com/mirth/connect/connectors/ws/DefaultAcceptMessage.java

I have imported ws-server from mirth at Java Build Path:
![imatge](https://user-images.githubusercontent.com/66680696/165257946-7b3a4468-f62e-4330-ac13-cafe796dfeee.png)

![imatge](https://user-images.githubusercontent.com/66680696/165258387-47133ad6-aa5f-41f2-a81d-caa7bab128e4.png)

This is the simple code:
````
package scc.ws;
import javax.jws.WebMethod;
import javax.jws.WebService;

import com.mirth.connect.connectors.ws.AcceptMessage;
import com.mirth.connect.connectors.ws.WebServiceReceiver;

@WebService
public class dadesDeRegistre extends AcceptMessage {

    public dadesDeRegistre(WebServiceReceiver webServiceReceiver) {
		super(webServiceReceiver);
		// TODO Auto-generated constructor stub
	}

    @WebMethod
    public String dadesRegistre(String message) {
        String response = webServiceReceiver.processData(message);

        if (response != null) {
            return response;
        }
        
        return null;
    }

}
````

What am I missing? Do you recommend doing it in eclipse?
Thanks

**Comment by ab-20-m** [comment, marked as answer]
You have a few options.

1. Change from a JRE 8 Runtime for your build path to a JDK 8. Note, if you switch to JDK 11, you will have to also do option 2.
2. Explicitly import the JAX-WS API JARs.

**Comment by FTK7** [comment]
Thanks!
I've done first option and it works!

I had tried the second one before, I downloaded the zip jaxws-ri-3.0.0.zip and imported some jars (I don't remember exactly which ones) and changed the imports to jakarta.jws.*. It compiled successfully and created the jars, but Mirth was giving the error "XXX has neither @WebService nor @WebServiceProvider annotation". I couldn't make it work.

Just for information, in Eclipse:
- Window ->Preferences
- Java -> Installed JREs
- Add -> Standard VM
- Write in JRE home the path to the installed jdk, in my case C:\Program Files\Java\jdk1.8.0_321\bin.
- Select it and apply changes.
